units/host_monitoring_idle_termination.go

```diff
@@ -0,0 +1,136 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	idleHostJobName = "idle-host-termination"
+
+	// idleTimeCutoff is the amount of time we wait for an idle host to be marked as idle.
+	idleTimeCutoff            = 4 * time.Minute
+	idleWaitingForAgentCutoff = 10 * time.Minute
+
+	// MaxTimeNextPayment is the amount of time we wait to have left before marking a host as idle
+	maxTimeTilNextPayment = 5 * time.Minute
+)
+
+type idleHostJob struct {
+	HostID     string `bson:"host" json:"host" yaml:"host"`
+	job.Base   `bson:"metadata" json:"metadata" yaml:"metadata"`
+	Terminated bool `bson:"terminated" json:"terminated" yaml:"terminated"`
+
+	env      evergreen.Environment
+	settings *evergreen.Settings
+	host     *host.Host
+}
+
+func makeIdleHostJob() *idleHostJob {
+	j := &idleHostJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    idleHostJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewIdleHostTerminationJob(env evergreen.Environment, h host.Host, id string) amboy.Job {
+	j := makeIdleHostJob()
+	j.host = &h
```

Comment:
Does this do anything if the host doesn't get persisted to the db?

Reply:
it only does things if the job doesn't get persisted to the db.